main.cpp

```diff
@@ -32,28 +32,286 @@ Purpose:  This project continues developing Project3.
+    std::string color { "white" };
+    std::string condition { "good" };
+    std::string securingMethod { "velcro" };
+    double percentOfSoleLeft = 100;
+    bool beingWorn = false;
+    
+    Shoe();
+    ~Shoe()
+    {
+        std::cout << "Shoe Destroyed\n";
+    }
+
+    void putOn ( std::string firstShoe = "R", bool leftFirst = true);
+    void takeOff ( std::string firstShoe = "R", bool leftFirst = true );
+    void throwAway ( bool donate = false );
+    void wearDown(int timesWorn);
+};
+
+Shoe::Shoe(){}
+
+void Shoe::putOn(std::string firstShoe, bool leftFirst) 
+{
+    if (leftFirst)
+    {
+        firstShoe = "L";
+    }
+}
+
+void Shoe::takeOff(std::string firstShoe, bool leftFirst) 
+{
+    if (leftFirst)
+    {
+        firstShoe = "L";
+    }
+}
 
+void Shoe::throwAway(bool donate) 
+{
+    if (condition == "good")
+    {
+        donate = true;
+    }
+    
+}
+
+void Shoe::wearDown(int timesWorn)
+{
+    for (int i = 0; i <= timesWorn; i += 1)
+    {
+        percentOfSoleLeft -= 0.2;
+    }
+
+    std::cout << "Sole percent left " << percentOfSoleLeft << std::endl;
+}
 /*
  copied UDT 2:
  */
+struct Wurlitzer
+{
+    int volumeLevel;
+    double vibratoAmount;
+    std::string model;
+    std::string condition;
+    int numKeys;
+    int output = 23;
+    bool on;
+    int tuning = 440;
+
+    Wurlitzer();
+    ~Wurlitzer()
+    {
+        std::cout << "Wurlitzer Destroyed\n";
+    }
+
+    void changeVolume ( int knobPosition );
+    void play ( int note = 23 );
+    void power ( bool switchPosition = true );
+    void printCondition();
+    int swellVolume(int amount, int time);
+};
+
+Wurlitzer::Wurlitzer() :
+volumeLevel ( 6 ),
+vibratoAmount ( .5 ),
+model ( "200a" ),
+condition ( "good" ),
+numKeys ( 49 ),
+output ( 23 ),
+on ( false )
+{
+
+}
+
+void Wurlitzer::changeVolume( int knobPosition ) 
+{
+    volumeLevel = knobPosition;
+    std::cout << "Knob Position is: " << knobPosition << "\n";
+}
+
+void Wurlitzer::play( int note ) 
+{
+    output = note;
+}
+
+void Wurlitzer::power( bool switchPosition ) 
+{
+    on = switchPosition;
+}
+
+void Wurlitzer::printCondition()
+{
+    std::cout << "This Wutlizer is in " << condition << " condition\n"; 
+}
 
+int Wurlitzer::swellVolume(int amount, int time)
+{
+    int swell = 0;
+    for (int i = 0; i <= time ; i += 1)
+    {
+        swell += 20;
+        if (swell >= amount)
+        {
+            std::cout << "Swell finished" << std::endl;
+            return swell;
+        }
+    }
+    return swell;
+}
 /*
  copied UDT 3:
  */
+struct ElectricBass
+{
+    int numStrings = 4;
+    std::string pickup = "precision";
+    std::string brand = "fender";
+    std::string model = "precision";
+    int yearMade = 1974;
+    int output = 23;
+    int cost = 1544;
+    int timeSinceRestrung = 13;
+    bool timeToRestring = false;
+    int tuning = 442;
+    
+    ElectricBass();
+    ~ElectricBass()
+    {
+        std::cout << "Wurlitzer Destroyed\n";
+    }
```

Comment:
0) move all of your implementations of all functions to OUTSIDE of the class. 
- [ ] revise